### ISSUE TITLE
Sharing iP code quality feedback [for @Dino-Nuggies]

### ISSUE THREAD

**Issue by soc-se-bot-blue** [Dino-Nuggies/ip#1]
@Dino-Nuggies We did an automated analysis of your code to detect potential areas to improve the code quality. We are sharing the results below, to help you improve the iP code further.

IMPORTANT: Note that the script looked for **just a few easy-to-detect problems only**, and **at-most three example are given** i.e., there can be other areas/places to improve.

### Aspect: Tab Usage

No easy-to-detect issues :+1: 

### Aspect: Naming boolean variables/methods

No easy-to-detect issues :+1: 

### Aspect: Brace Style

No easy-to-detect issues :+1: 

### Aspect: Package Name Style

No easy-to-detect issues :+1: 

### Aspect: Class Name Style

No easy-to-detect issues :+1: 

### Aspect: Dead Code

No easy-to-detect issues :+1: 

### Aspect: Method Length

Example from [`src/main/java/lemon/Lemon.java` lines `40-181`](https://github.com/Dino-Nuggies/ip/blob/4bea401ac64ae95f64b8cc97bf4f69fd2cf8d343/src/main/java/lemon/Lemon.java#L40..L181):

```java

    public void runLemon() {
        if (!isInitialised) {
            System.out.print(" Ouhiiee, my head hurrrtt, i dont think im initialized properly ;-;\n"
                    + " Please try again...");
            return;
        }

        boolean isRunning = true;

        ui.printIntroMsg();

        while (isRunning) {
            ui.printBarMsg();

            String input = systemScanner.next().toUpperCase();
            ui.printBarMsg();
            try {
                Parser.Commands command = Parser.Commands.valueOf(input);
                switch (command) {
                case BYE: {
                    boolean isAbleToSave = storage.saveTasks(tasks);
                    if (isAbleToSave) {
                        isRunning = false;
                        ui.printEndingMsg();
                    }
                    break;
                }
                case LIST:
                    if (tasks.size() == 0) {
                        ui.printEmptyListMsg();
                    } else {
                        ui.printListMsg(tasks.toString());
                    }
                    break;
                case MARK: {
                    int next = systemScanner.nextInt();
                    if (next > tasks.size() || next <= 0) {
                        throw new InvalidCommandException(" OOPS!!! Please select a valid task");
                    }

                    ui.printMarkMsg();
                    tasks.get(next - 1).markDone();

                    System.out.println("   " + tasks.get(next - 1).toString());
                    break;
                }
                case UNMARK: {
                    int next = systemScanner.nextInt();
                    if (next > tasks.size() || next <= 0) {
                        throw new InvalidCommandException(" OOPS!!! Please select a valid task");
                    }

                    ui.printUnmarkMsg();
                    tasks.get(next - 1).unmarkDone();

                    System.out.println("   " + tasks.get(next - 1).toString());
                    break;
                }
                case TODO: {
                    String next = systemScanner.nextLine();

                    Task todo = new Todo(next, false);
                    tasks.addNewTask(todo);
                    ui.printAddTaskMsg(todo.toString(), tasks.size());
                    break;
                }
                case DEADLINE: {
                    String[] next = systemScanner.nextLine().split("/by ");
                    if (next.length < 2) {
                        throw new InvalidFormatException(" Missing/incorrect date format \n"
                                + " Use /by and format date as dd-mm-yyyy");
                    }

                    LocalDate by = LocalDate.parse(next[1], DateTimeFormatter.ofPattern("dd-MM-yyyy"));

                    Task deadline = new Deadline(next[0], by, false);
                    tasks.addNewTask(deadline);
                    ui.printAddTaskMsg(deadline.toString(), tasks.size());
                    break;
                }
                case EVENT: {
                    String[] next = systemScanner.nextLine().split("/from | /to ");

                    if (next.length < 3) {
                        throw new InvalidFormatException(" Missing/incorrect date format \n"
                                + " Use /by and format date as dd-mm-yyyy");
                    }

                    LocalDate from = LocalDate.parse(next[1], DateTimeFormatter.ofPattern("dd-MM-yyyy"));
                    LocalDate to = LocalDate.parse(next[1], DateTimeFormatter.ofPattern("dd-MM-yyyy"));

                    Task event = new Event(next[0], from, to, false);
                    tasks.addNewTask(event);
                    ui.printAddTaskMsg(event.toString(), tasks.size());
                    break;
                }
                case DELETE: {
                    int next = systemScanner.nextInt();
                    if (next > tasks.size() || next <= 0) {
                        throw new InvalidCommandException(" OOPS!!! Please select a valid task");
                    }

                    Task t = tasks.deleteTask(next);
                    ui.printDeleteTaskMsg(t.toString(), tasks.size());
                    break;
                }
                case FIND: {
                    String next = systemScanner.nextLine();
                    if (next.isEmpty() || next.equals(" ")) {
                        throw new InvalidFormatException(" Please specify what you need me to find ^-^");
                    }

                    TaskList matchingTasks = tasks.findTasks(next);
                    if (matchingTasks.size() > 0) {
                        ui.printMatchingTaskMsg(matchingTasks.toString());
                    } else {
                        ui.printNoMatchingMsg();
                    }
                    break;
                }
                default:
                    throw new IllegalArgumentException();
                }
            } catch (InvalidCommandException | DescriptionException e) {
                System.out.println(e.getMessage());
                systemScanner.nextLine();
            } catch (InputMismatchException e) {
                System.out.println(" OOPS!!! I cant understand your mark/unmark, please select using numbers xd");
                systemScanner.nextLine();
            } catch (IllegalArgumentException e) {
                System.out.println(" OOPS!!! I'm sowwy, but I don't know what that means :-(\n\n"
                        + " I can help you add tasks with \"todo\", \"deadline\", \"event\"\n"
                        + " I can also keep track of all your tasks with \"list\"\n"
                        + " If you wanna update certain tasks, use \"mark\" or \"unmark\" and then its number");
                systemScanner.nextLine();
            } catch (DateTimeParseException e) {
                System.out.println(" Incorrect date format, make sure the format is dd-MM-yyyy");
            } catch (InvalidFormatException e) {
                System.out.println(e.getMessage());
            }
        }
    }

``` 

**Suggestion:** Consider applying [SLAP (and other abstraction mechanisms)](https://nus-cs2103-ay2425s1.github.io/website//se-book-adapted/chapters/codeQuality.html#guideline-maximise-readability) to shorten methods e.g., extract some code blocks into separate methods. You may ignore this suggestion if you think a longer method is justified in a particular case.

### Aspect: Class size

No easy-to-detect issues :+1: 

### Aspect: Header Comments

Example from [`src/main/java/lemon/Lemon.java` lines `37-39`](https://github.com/Dino-Nuggies/ip/blob/4bea401ac64ae95f64b8cc97bf4f69fd2cf8d343/src/main/java/lemon/Lemon.java#L37..L39):

```java

    /**
     * Execute lemon once initialization is done
     */

```

Example from [`src/main/java/lemon/TaskList.java` lines `23-27`](https://github.com/Dino-Nuggies/ip/blob/4bea401ac64ae95f64b8cc97bf4f69fd2cf8d343/src/main/java/lemon/TaskList.java#L23..L27):

```java

    /**
     * Add a new task to arraylist
     * @param t task that is added into the list
     * @throws DescriptionException Exception when the description of the task is empty
     */

```

Example from [`src/main/java/lemon/TaskList.java` lines `40-44`](https://github.com/Dino-Nuggies/ip/blob/4bea401ac64ae95f64b8cc97bf4f69fd2cf8d343/src/main/java/lemon/TaskList.java#L40..L44):

```java

    /**
     * Remove a task from the arraylist
     * @param i index of the task within the list
     * @return task that is removed
     */

``` 

**Suggestion:** Ensure method/class header comments follow the format specified in the coding standard, in particular, the phrasing of the overview statement.

### Aspect: Recent Git Commit Message

possible problems in commit [`50808dc`](https://github.com/Dino-Nuggies/ip/commit/50808dcb9b34145b23995b994b195416fe6fb06f):

```

Add find

Use find as a way to find a task by searching for a keyword in the task description.


```

* body not wrapped at 72 characters: e.g., `Use find as a way to find a task by searching for a keyword in the task description.`
             

**Suggestion:** Follow the given [conventions for Git commit messages](https://se-education.org/guides/conventions/git.html) for _future_ commits (do not modify past commit messages as doing so will change the commit timestamp that we used to detect your commit timings).

### Aspect: Binary files in repo

No easy-to-detect issues :+1: 

----

:information_source: The bot account used to post this issue is un-manned. **Do not reply to this post** (as those replies will not be read). Instead, contact `cs2103@comp.nus.edu.sg` if you want to follow up on this post.
